src/pybind/mgr/dashboard/tests/test_osd.py

```diff
@@ -374,3 +374,48 @@ def test_reweight_osd(self, instance):
         self._task_post('/api/osd/1/reweight', {'weight': '1'})
         instance.send_command.assert_called_with('mon', 'osd reweight', id=1, weight=1.0)
         self.assertStatus(200)
+
+    @mock.patch('dashboard.controllers.orchestrator.OrchClient.instance')
+    def test_deployment_options(self, instance):
+        fake_client = mock.Mock()
+        instance.return_value = fake_client
+        fake_client.get_missing_features.return_value = []
+
+        class MockDevice:
+            def __init__(self, human_readable_type, path, available=True):
+                self.human_readable_type = human_readable_type
+                self.available = available
+                self.path = path
+
+        def create_invetory_host(devices_data):
+            inventory_host = mock.Mock()
+            inventory_host.devices.devices = []
+            for data in devices_data:
+                inventory_host.devices.devices.append(MockDevice(data['type'], data['path']))
+            return inventory_host
+
+        devices_data = [
+            {'type': 'hdd', 'path': '/dev/sda'},
+            {'type': 'hdd', 'path': '/dev/sdb'},
+            {'type': 'hdd', 'path': '/dev/sdc'},
+            {'type': 'hdd', 'path': '/dev/sdd'},
+            {'type': 'hdd', 'path': '/dev/sde'},
+        ]
+        inventory_host = create_invetory_host(devices_data)
+        fake_client.inventory.list.return_value = [inventory_host]
+        self._get('/api/osd/deployment_options')
+        self.assertStatus(200)
+        res = self.json_body()
+        self.assertTrue(res['options']['cost-capacity']['available'])
+        assert res['recommended_option'] == 'cost-capacity'
```

Comment:
@pereman2 Can you please cover tests related to other two options as well i.e `throughput-optimized` and `iops-optimized`?